sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -111,14 +116,52 @@ class AccountState(val params: NetworkParams,
       stateView.applyMainchainBlockReferenceData(mcBlockRefData).get
     }
 
-    for (tx <- mod.sidechainTransactions) {
-      stateView.applyTransaction(tx).get
+    // get also list of receipts, useful for computing the receiptRoot hash
+    val receiptList = new ListBuffer[EthereumReceipt]()
+    val blockNumber = stateView.getHeight
+    val blockHash = idToBytes(mod.id)
+    var cumGasUsed : BigInteger = BigInteger.ZERO
+
+    for ((tx, txIndex) <- mod.sidechainTransactions.zipWithIndex) {
+      stateView.applyTransaction(tx, cumGasUsed) match {
+        case Success(consensusDataReceipt) =>
+          val txGasUsed = consensusDataReceipt.cumulativeGasUsed.subtract(cumGasUsed)
+          // update cumulative gas used so far
+          cumGasUsed = consensusDataReceipt.cumulativeGasUsed
+          val ethTx = tx.asInstanceOf[EthereumTransaction]
+
+          val txHash = idToBytes(ethTx.id)
+
+          val contractAddress = if (ethTx.getTo != null && !stateView.isEoaAccount(ethTx.getTo.address()) ) {
+            ethTx.getTo.address()
+          } else {
+            new Array[Byte](Address.LENGTH)
+          }
+
+          // get a receipt obj with non consensus data too (logs updated too)
+          val fullReceipt = EthereumReceipt(consensusDataReceipt,
+                      txHash, txIndex, blockHash, blockNumber, txGasUsed, contractAddress).updateLogs()
```

Comment:
I think that we can avoid calling the `.updateLogs()` constructor is enough.
Receipt should contain only a `EthereumConsensusDataLog` and have a lazy method to get the `EthereumLog`

Reply:
Ok, done